solr/core/src/java/org/apache/solr/cli/ToolBase.java

```diff
@@ -48,9 +52,30 @@ protected void echo(final String msg) {
     stdout.println(msg);
   }
 
+  @Override
+  public Options getAllOptions() {
+    return new Options()
+        .addOption(CommonCLIOptions.HELP_OPTION)
+        .addOption(CommonCLIOptions.VERBOSE_OPTION);
+  }
+
+  @Override
+  public Options getOptions() {
+    Options options = new Options();
+
+    Collection<Option> toolOpts = getAllOptions().getOptions();
+    for (Option toolOpt : toolOpts) {
+      if (!toolOpt.isDeprecated()) {
+        options.addOption(toolOpt);
+      }
+    }
+    return options;
+  }
+
   @Override
   public int runTool(CommandLine cli) throws Exception {
-    verbose = cli.hasOption(SolrCLI.OPTION_VERBOSE.getLongOpt());
+    verbose = cli.hasOption(CommonCLIOptions.VERBOSE_OPTION);
```

Comment:
This does read nicer!

Reply:
I would probably prefer the option group to be placed in `CommonCLIOptions` as another "option", mainly because of context, class responsibilities and for later referencing in other classes. I believe that there won't be any overrides either in implementations of `ToolBase`.

```java
// If CommonCLIOptions only imported
super.getOptions()
    .addOptionGroup(CommonCLIOptions.CONNECTION_OPTIONS)

...

if (cli.hasOption(CommonCLIOptions.CONNECTION_OPTIONS) { ... }
```

`getConnectionOptions` being placed in `ToolBase` reminds me on `getOptions()` and I would expect `Options` to be returned, rather than an `OptionGroup`.